### ISSUE TITLE
fetch + signal memory leak

### ISSUE THREAD

**Issue by exe-dealer** [denoland/deno#16487]
deno v1.27.0

```javascript
const { signal } = new AbortController();
const kAbortAlgos = Object.getOwnPropertySymbols(signal).find(s => String(s) == 'Symbol([[abortAlgos]])');

console.log('abortAlgos size before', signal[kAbortAlgos]?.size);
for (let i = 0; i < 100; i++) {
  const resp = await fetch('data:,', { signal });
  await resp.arrayBuffer();
}
console.log('abortAlgos size after', signal[kAbortAlgos].size);
```

stdout:
```
abortAlgos size before undefined
abortAlgos size after 100
```

**Comment by exe-dealer** [comment]
https://github.com/denoland/deno/blob/dbcbf53ab5168ad9c5d6ca8b4c7740db98b270a6/ext/web/03_abort_signal.js#L182-L183

this callback is added to signal but never removed. If signal lifetime is the same as application lifetime, then memory usage grows until deno crash.

**Comment by nktpro** [comment]
I could also confirm this issue still persists as of the latest `1.33.2` release.

**Comment by lucab** [comment]
The `fetch` spec related to this has been updated by https://github.com/whatwg/fetch/pull/1646 in the meanwhile, so the proper fix for this will likely follow after https://github.com/denoland/deno/issues/18944 is done.

**Comment by lucab** [comment]
Fixed as part of #21087.
